### DISCUSSION TITLE
Failed Password Prompt Matching

### DISCUSSION THREAD

**Discussion by NthNEK** [scrapli/scrapli_community#152] [Q&A] [answered]
Hi,

Just started using scrapli today and I've read for hours, nearly everything I can find and didn't find the answer so I apologize if this is documented somewhere... I have some devices that provide a specialized password prompt (from TACACS+) that scrapli was unable to detect (such as seen in the following relevant logs):

```
DEBUG:scrapli.channel:read: b'(<user redacted>@<host redacted>) Password <other words redacted>: '
CRITICAL:scrapli.channel:operation timed out, closing connection
```

Since I'm using the standard core transport `SystemSSH`, I found and modified line 114 in `scrapli/channel/base_channel.py`:

From:
```
self._auth_password_pattern = r"(.*@.*)?password:\s?$"
```
To:
```
self._auth_password_pattern = r"(.*@.*)?password.*:\s?$"
```
...just adding in the `.*` between `password` and `:` as a test. This fixed my issue for testing purposes, but I'm wondering if there's a better way to handle altering the password prompt that I'm missing? I'd prefer to use a method that won't be broken upon updating scrapli from pip/source.

**Comment by carlmontanari** [comment, marked as answer]
there is a property w/ a setter on the channel, so I guess that would be the "recommended" way to change this. that is [here](https://github.com/carlmontanari/scrapli/blob/25fab4873eb0308e9d67787a513ca6ad195597d0/scrapli/channel/base_channel.py#L157-L195)

so something like `my_conn.channel.auth_password_pattern = blah` should do the trick

**Comment by NthNEK** [comment]
A much faster answer than I expected. Thanks for all your work on this, Carl! It is truly appreciated.

My tired eyes yesterday didn't even realize I was posting this question in the scrapli_community discussions section when I thought I was in nornir_scrapli.
